runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaPublishFactory.java

```diff
@@ -163,6 +172,7 @@ private final class MqttPublishProxy
         private final KafkaRetainedProxy retained;
         private final String16FW kafkaMessagesTopic;
         private final String16FW kafkaRetainedTopic;
+        private final List<Function<String, String>> clients;
```

Comment:
We should be storing `kafkaMessagesTopic` on `KafkaMessagesProxy` and `kafkaRetainedTopic` on `KafkaRetainedProxy` instead.